src/main/java/com/m2gi/movieMarket/domain/entity/cart/Cart.java

```diff
@@ -0,0 +1,29 @@
+package com.m2gi.movieMarket.domain.entity.cart;
+
+import java.io.Serializable;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Set;
+import java.util.Collection;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name="cart")
+public class Cart implements Serializable {
+    @Id
+    @Column(name="cart_id")
+    @GeneratedValue(strategy=GenerationType.AUTO)
+    private int id;
+
+    @OneToMany(mappedBy = "cart")
+    private Collection<CartDetail> cartDetailCollection;
```

Comment:
Sur les autres associations `OneToMany` que je suis entrain de faire j'ai plutôt fais
```java
@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
@JoinColumn(name = "user_id")
private List<UserRole> userRoles = new ArrayList<>();
```
Le `JoinColumn` te permet de créer une association unidirectionnel sans table intermédiaire.
Le `cascade` est important ici je pense.
Voir si on ajoute des infos sur le `fetch`.
Concernant le type de l'attribut, j'ai opté pour  un `ArrayList`, tu as des arguments concernant l'utilisations d'une `Collection` plutôt que un `ArrayList` ?

Reply:
La Collection était le type abstrait, tu as tout bon pour l'ArrayList
Plus d'infos sur ce [thread](https://stackoverflow.com/questions/21974361/what-java-collection-should-i-use)

Reply:
comment : Hibernate demande une interface d'où l'utilisation du List<>
[infos](http://docs.jboss.org/hibernate/core/3.3/reference/en/html/collections.html)